devlib/target.py

```diff
@@ -736,12 +736,43 @@ def batch_revertable_write_value(self, kwargs_list):
 
     def write_value(self, path, value, verify=True):
         value = str(value)
-        self.execute('echo {} > {}'.format(quote(value), quote(path)), check_exit_code=False, as_root=True)
+
         if verify:
-            output = self.read_value(path)
-            if not output == value:
-                message = 'Could not set the value of {} to "{}" (read "{}")'.format(path, value, output)
+            # Check in a loop for a while since updates to sysfs files can take
+            # some time to be observed, typically when a write triggers a
+            # lengthy kernel-side request, and the read is based on some piece
+            # of state that may take some time to be updated by the write
+            # request, such as hotplugging a CPU.
+            cmd = '''
+orig=$(cat {path} 2>/dev/null || printf "")
+printf "%s" {value} > {path} || exit 10
+if [ {value} != "$orig" ]; then
+   trials=0
+   while [ "$(cat {path} 2>/dev/null)" != {value} ]; do
+       if [ $trials -ge 10 ]; then
+           cat {path}
+           exit 11
+       fi
+       sleep 0.01
+       trials=$((trials + 1))
+   done
+fi
+'''
+        else:
+            cmd = 'printf "%s" {value} > {path}'
```

Comment:
Not all targets may have `printf`, use the busybox implementation for consistency?

Reply:
I always forget that we don't put busybox symlinks in PATH, will do